core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java

```diff
@@ -579,6 +579,12 @@ public static SqlOperandTypeChecker variadic(
           .or(OperandTypes.family(SqlTypeFamily.MAP))
           .or(OperandTypes.family(SqlTypeFamily.ANY));
```

Comment:
ARRAY_OR_MAP is no where used, can we just remove it

Reply:
yes , right! but we can keep it as of now since its a valid operand type and may come handy for ARRAY or MAP.